stingray/largememory.py

```diff
@@ -0,0 +1,558 @@
+        Do not show a progress bar when generating an averaged cross spectrum.
+        Useful for the batch execution of many spectra
+    dt1: float
+        The time resolution of the light curve. Only needed when constructing
+        light curves in the case where data1 or data2 are of :class:EventList
+
+    Returns
+    -------
+    object
+        Summed computed spectra.
+
+    Raises
+    ------
+    ValueError
+        If spectra is not AveragedCrossspectrum or AveragedPowerspectrum
+    """
+    times = zarr.open_array(store=data_path[0], mode='r', path='times')
+
+    dt = zarr.open_array(store=data_path[1], mode='r', path='dt')
+    ncounts = zarr.open_array(store=data_path[1], mode='r', path='ncounts')
+    mjdref = zarr.open_array(store=data_path[1], mode='r', path='mjdref')
+
+    try:
+        energy = zarr.open_array(store=data_path[0], mode='r', path='energy')
+    except ValueError:
+        energy = None
+
+    try:
+        gti = zarr.open_array(store=data_path[0], mode='r', path='gti')
+    except ValueError:
+        gti = None
+
+    try:
+        pi_channel = zarr.open_array(store=data_path[0],
+                                     mode='r',
+                                     path='pi_channel')
+    except ValueError:
+        pi_channel = None
+
+    if spec_type == 'AveragedPowerspectrum':
+        fin_spec = stingray.AveragedPowerspectrum()
+
+    elif spec_type == 'AveragedCrossspectrum':
+        times_other = zarr.open_array(store=data_path[2],
+                                      mode='r',
+                                      path='times')
+
+        dt_other = zarr.open_array(store=data_path[3], mode='r', path='dt')
+        ncounts_other = zarr.open_array(store=data_path[3],
+                                        mode='r',
+                                        path='ncounts')
+        mjdref_other = zarr.open_array(store=data_path[3],
+                                       mode='r',
+                                       path='mjdref')
+
+        try:
+            energy_other = zarr.open_array(store=data_path[2],
+                                           mode='r',
+                                           path='energy')
+        except ValueError:
+            energy_other = None
+
+        try:
+            gti_other = zarr.open_array(store=data_path[2],
+                                        mode='r',
+                                        path='gti')
+        except ValueError:
+            gti_other = None
+
+        try:
+            pi_channel_other = zarr.open_array(store=data_path[2],
+                                               mode='r',
+                                               path='pi_channel')
+        except ValueError:
+            pi_channel_other = None
+
+        fin_spec = stingray.AveragedCrossspectrum()
+
+    else:
+        raise ValueError
+
+    flag = True
+    for i in range(times.chunks[0], times.size, times.chunks[0]):
+        ev1 = EventList(
```

Comment:
note that this will not work: while light curves have a fixed number of bins in a given amount of time, the number of events in an event list are stochastic. To be consistent, you should decide the light curve segment size and read all the event lists that fall into a given segment of light curve. Can zarr do it?

Reply:
So essentially we take the segment size from the user and read all the events that fall in that segment?

Reply:
Yes

Reply:
What we could do is using the gtis and the size of each chunk on disk read data of a specified size say 10MB, which may span over multiple chunks and create a EventList

Reply:
There's `stingray.gti.time_intervals_from_gtis` that gives you time limits strictly inside GTIs given the segment size

Reply:
I tried doing this, but I guess I am missing a piece of the puzzle. Could you help me with this?